airflow/www/decorators.py

```diff
@@ -125,6 +126,9 @@ def wrapper(*args, **kwargs):
                     extra_fields = {**extra_fields, **masked_json}
 
                 params = {**request.values, **request.view_args}
+                
+                dag_id = params.get("dag_id") or session.scalar(select(DagRun.dag_id)
+                                                                .where(DagRun.id == params.get("pk")))
```

Comment:
What if we did something like this instead to avoid the issue mentioned earlier?
```suggestion
                dag_id = params.get("dag_id") or (session.scalar(select(DagRun.dag_id)
                                                                .where(DagRun.id == params.get("pk"))) if event_name == "dagrun.delete" else None)

```

Reply:
This allows me to still delete taskinstances fine and also dagruns.

<img width="762" alt="image" src="https://github.com/apache/airflow/assets/9200263/a5292696-d2f3-4f47-b604-bb93ad35e888">

However, if I tried to delete multiple, this would still not work -

<img width="948" alt="image" src="https://github.com/apache/airflow/assets/9200263/ff9048dc-e80b-4dfc-9ca8-7938b132bd1e">

<img width="945" alt="image" src="https://github.com/apache/airflow/assets/9200263/14de6dd2-762f-495c-9abe-4984648a1242">